src/libs/state/BufferState.js

```diff
@@ -61,6 +62,25 @@ export default class BufferState {
         if (this.isChannel()) {
             maybeStartWhoLoop(this);
         }
+
+        function onNetworkConnecting(event) {
+            if (event.network === this.getNetwork()) {
+                this.chathistory_request_count = 0;
+            }
+        }
+
+        function onBufferClose(event) {
+            if (event.buffer === this) {
+                this.state.$off('network.connecting', onNetworkConnectingBound);
+                this.state.$off('buffer.close', onBufferCloseBound);
+            }
+        }
+
+        const onNetworkConnectingBound = onNetworkConnecting.bind(this);
+        const onBufferCloseBound = onBufferClose.bind(this);
+
+        state.$on('network.connecting', onNetworkConnectingBound);
+        state.$on('buffer.close', onBufferCloseBound);
```

Comment:
These 2 event handlers could do with some comments explaining why they're here, it's not obvious for someone glancing at it